strr-web/components/bcros/form-section/property/Address.vue

```diff
@@ -84,57 +143,107 @@
 
 <script setup lang="ts">
 import { CountryItem } from '@/interfaces/address-i'
+import { PropertyTypeValueMapE } from '@/enums/property-type-map-e'
 import countries from '@/utils/countries.json'
 const { t } = useTranslation()
+const unitNumberPlaceholder = ref('')
 
 const country = defineModel<string>('country')
-const address = defineModel<string>('address')
+const streetNumber = defineModel<string>('streetNumber')
+const streetName = defineModel<string>('streetName')
+const unitNumber = defineModel<string>('unitNumber')
 const addressLineTwo = defineModel<string>('addressLineTwo')
 const city = defineModel<string>('city')
 const province = defineModel<string>('province')
 const postalCode = defineModel<string>('postalCode')
 const nickname = defineModel<string>('nickname')
 const countryItems = ref<CountryItem[]>([])
-const fieldErrors = ref({
-  addressLineTwo: '',
-  city: '',
-  postalCode: ''
-})
 
-const addressComplete = () => {
+const addressComplete = (initiatedFromStreetName: boolean) => {
   if (typeof country.value === 'string') {
-    enableAddressComplete(id, 'CA', false)
+    if (initiatedFromStreetName) {
+      enableAddressComplete(streetNameId, 'CA', false, 'BC')
+    } else {
+      enableAddressComplete(streetNumberId, 'CA', false, 'BC')
+    }
   }
 }
 
-const resetFieldError = (field: keyof typeof fieldErrors.value) => {
-  fieldErrors.value[field] = ''
-}
-
-const onAddressInput = () => {
+const onAddressInput = (initiatedFromStreetName: boolean) => {
+  if (!initiatedFromStreetName) {
+    streetName.value = ''
+  }
+  unitNumber.value = ''
   addressLineTwo.value = ''
   city.value = ''
   postalCode.value = ''
-  resetFieldError('addressLineTwo')
-  resetFieldError('city')
-  resetFieldError('postalCode')
-  addressComplete()
+  addressComplete(initiatedFromStreetName)
 }
 
 const {
-  id,
+  streetNumberId,
+  streetNameId,
   defaultCountryIso2,
   enableAddressComplete,
-  addressNotInBC
+  addressNotInBC,
```

Comment:
Why not pass a single error object like before, instead of one by one... If you feel like doing small updated, you can refactor, otherwise leave to me for my validation updates.